src/defrag.c

```diff
@@ -1275,10 +1278,10 @@ void *activeDefragHExpiresOB(void *ptr, void *privdata) {
         serverAssert(exlink != NULL);
     }
 
+    link = kvstoreDictFindLink(db->keys, slot, keystr, NULL);
```

Comment:
this dictFind could be excessive (if the `kvobj`) isn't moved,  we don't have to fix it.
maybe we can use here the re-introduced dictFindByHashAndPtr too?

Reply:
Another option is to activeDefragAlloc() and if reallocated, then call a new function dictReplaceByPtr() that will get:
*  key string to lookup, (extracted the reallocated kvobj) 
* The old pointer 
* The new pointer

Reply:
@moticless its name should be dictReplaceKeyByPtr. If we want to replace the value, we also need a dictReplaceValueByPtr. From this perspective, dictFindByHashAndPtr would be more universal.

Reply:
@oranagra @moticless please take a look this commit: https://github.com/redis/redis/pull/14323/commits/67c7b74298a58294ab2e649769c6b1840400f0ab

Using a two-stage reallocation and release feels the code will be simpler.